utils/types/responses.ts

```diff
@@ -0,0 +1,35 @@
+import { Attribute, Category, InventoryItem, ItemDefinition, User } from '.'
+
+export interface UserResponse extends User {
+  _id: string
+}
+
+export interface ItemDefinitionResponse extends ItemDefinition {
+  _id: string
+  category?: Category
+  attributes?: Attribute[]
+}
+
+export interface InventoryItemResponse extends InventoryItem {
+  _id: string
+  itemDefinition: string
+  attributes?: {
+    attribute: Attribute
```

Comment:
when we return objects within objects like this, it should use the response object. so here we would use 'AttributeRequest`

Reply:
good catch, fixed